src/server.rs

```diff
@@ -13,6 +13,8 @@ pub enum ServerErrorKind {
     /// Failed to bind to a port.
     #[fail(display = "failed to bind to the port")]
     Bind,
+    #[fail(display = "failed to create process pool")]
+    Spawn,
```

Comment:
Can we give this a different name and description? Later on, there might be other errors thrown in `ServiceState::Create` that are unrelated to process pools.